src/Raven.Server/Documents/DocumentDatabase.cs

```diff
@@ -508,7 +502,8 @@ private void ExecuteClusterTransaction()
                     continue;
                 }
 
-                _hasClusterTransaction.Wait(DatabaseShutdown);
+                //To make sure we mark compare exchange tombstone for cleaning  
+                _hasClusterTransaction.Wait(TimeSpan.FromMinutes(5), DatabaseShutdown);
```

Comment:
In the new implementation, we use compare exchange tombstones to mark the last cluster transaction index. 
The main goal is to update the "last cluster transaction" only in DocumentDatabase.ExecuteClusterTransaction to avoid bugs like the ones we had in the related issue. 
We need to update this for both cases: when there are document commands in the cluster transaction and when there are not.

If there are document commands, the solution is straightforward: we update while handling the document commands.

To handle the case when the cluster transaction has no document commands, we decided to rely on the last compare exchange, including its tombstone. 
We don't differentiate between compare exchange from a cluster transaction or compare exchange from a regular operation. 
Therefore, we need to subscribe to the compare exchange tombstone cleaner. 
Cleaning the tombstone before we update the last cluster-wide transaction index can cause future requests to hang.

The issue that arises here is what happens if the database doesn't have cluster-wide transactions at all, and the system creates and deletes compare exchange by regular operations. 
We still need to mark the tombstone for cleaning. 
To address this, I added a timeout of 5 minutes, which forces the thread to reach the code where it marks the tombstone for cleaning.

I wonder what the right approach should be:

1. Keep the hard-coded 5 minutes.
2. Use the Cluster.CompareExchangeTombstonesCleanupIntervalInMin configuration.
3. Create a dedicated configuration.

Reply:
I'm fine with that

Reply:
Let's create a dedicated configuration for this.

Reply:
Done.
I tried to find a well-described name for it but it became a little bit long.
`MaxClusterTransactionCompareExchangeTombstoneCheckInterval`